src/components/Form.jsx

```diff
@@ -1,17 +1,20 @@
 import React, { useRef, useState } from 'react';
+import { useDispatch } from 'react-redux';
 import { v4 as uuidv4 } from 'uuid';
 import PropTypes from 'prop-types';
 import SelectCategory from './SelectCategory';
+import { addBook } from '../redux/book/bookSlice';
 import { categories } from '../constants/constants';
 
-function Form({ setBooks }) {
-  const [selected, setSelected] = useState(categories[0]);
+function Form({ onAddBook }) {
+  const [title, setTitle] = useState('');
+  const [author, setAuthor] = useState('');
```

Comment:
- [ ] [OPTIONAL] you can actually use one state to manage all your input values like 
```
 const [input, setInput] = useState({
author: '',
title: ''
});
```